### PR TITLE
feat(scripts): add computing line of code python script

### PR DESCRIPTION
**Pull request by LinerSu**
Add a python script to compute all lines of code info based on each harness that Seahorn analyzes.
The script work when you build all harnesses already.
You can set arg `--debug` to display detailed loc which computed based on each function call.

### CODE REVIEW

File: scripts/get_loc_info.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+     # We treat declare function as one line of code, 
+     #      those function will be either simplified code or predefined code on seahorn
+     # The function recursively compute each called function on the harness
+     # Then it collect loc of current function and store it into total_loc
+     # @param data stored data from .ll file.
+     # @param name the function name we used
+     # @param total_loc the result of loc that we computed
+     # @return nothing, use call by ref.
+    """
+    name_regexp, _ = get_regex_based_on_func_name(name)
```
Comment (danblitzhou) [inline scripts/get_loc_info.py:130]: judging from the debug message, we seem to count `loc` of a same function multiple times if its called multiple times, is that the behaviour we are going for? if not, we can use a dictionary of {name: loc} instead and tally up loc of each function exactly once.

File: scripts/get_loc_info.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+import re
+import os
+import csv
+import glob
+import math
+import subprocess
+import argparse
+
+# Args for this script
+parser = argparse.ArgumentParser(description='Present debug flag if using debug mode.')
```
Comment (danblitzhou) [inline scripts/get_loc_info.py:11]: nit: maybe move this into `__main__`?

File: scripts/get_loc_info.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+parser.add_argument('--debug', action='store_true', default=False)
+args=parser.parse_args()
+
+builddir = '../build/'
+buildabspath = os.path.abspath(builddir)
+dataabspath = os.path.abspath('../') + "/data"
+harness_path = f'{buildabspath}/seahorn/jobs'
+benchsubdirs = sorted(next(os.walk(f'{builddir}/seahorn/jobs'))[1])
+# benchsubdirs = ['add_size_checked', 'array_list_back']
```
Comment (danblitzhou) [inline scripts/get_loc_info.py:20]: nit pick: remove please

File: scripts/get_loc_info.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+import math
+import subprocess
+import argparse
+
+# Args for this script
+parser = argparse.ArgumentParser(description='Present debug flag if using debug mode.')
+parser.add_argument('--debug', action='store_true', default=False)
+args=parser.parse_args()
+
+builddir = '../build/'
```
Comment (danblitzhou) [inline scripts/get_loc_info.py:15]: nit pick: capitalize these constants

File: scripts/get_loc_info.py [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+        if name_regexp.search(plain_d):
+            if 'declare' in plain_d:  # no definition, assume one line code
+                total_loc[0] += 1
+                break
+            while True:
+                call_name, debug_name, index = get_next_func_call(
+                    data, name, index + 1)
+                if not call_name and not debug_name:
+                    break
+                # print(name, call_name, debug_name)
```
Comment (danblitzhou) [inline scripts/get_loc_info.py:141]: nit pick: remove or conditionally enable during debug mode

File: scripts/get_loc_info.py [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+    try:
+        file = open(file_name, 'r')
+    except BaseException:
+        sys.exit("human readable bitcode (.ll) file does not exits. Exit!")
+    data = file.readlines()
+    if args.debug:
+        print(f'start compute loc for harness <{bench_name}>')
+    main_regexp, main_debug_regexp = get_regex_based_on_func_name('main')
+    res_data = ['', '']
+    plain_data = data
```
Comment (danblitzhou) [inline scripts/get_loc_info.py:160]: nit pick: I don't think plain_data is ever used in this function

**Review by danblitzhou — COMMENTED** [review]
In general looks good; tested locally and looked at a few examples, loc of individual functions seem to be correct; not sure if over-counting multiple invocation of same function is what we want though.

File: scripts/get_loc_info.py (hunk 1 above)
Comment (LinerSu) [reply scripts/get_loc_info.py:130]: Maybe I need to discuss this with arie later. This is what I understand for now. I think the loc of each harness that we analyze is all the functions we used/called in the running time.

Comment (LinerSu) [reply scripts/get_loc_info.py:130]: I think you are right. I will change to avoid counting the same function multiple times.

**Review by danblitzhou — APPROVED** [review]
LGTM

**Pull request merged**